native/src/modules/layout/containers/HeaderContainer.js

```diff
@@ -67,8 +103,10 @@ const mapStateToProps = (state: StateType, ownProps: OwnPropsType): StatePropsTy
     })
   }
   const peeking = isPeekingRoute(state, { routeCity: route.city })
+  const shareUrlFromScene = ownProps.scene.route.params?.shareUrl
+  const shareUrl: any = getShareUrl(state.cityContent, route, routeKey, routeCityModel, shareUrlFromScene)
```

Comment:
Together with the suggestion above this should work:
```suggestion
   // $FlowFixMe shareUrl is a string if set
  const shareUrlFromScene: string | null = ownProps.scene.route.params?.shareUrl || null
  const { city, language } = route
  const path = route.path || null
  const shareUrl = shareUrlFromScene || getShareUrl({ city, language, path, routeName: ownProps.scene.route.name })
```
You have to remove the shareUrl from the categories, pois, news and events routes though in `NavigationTypes.js` (which you should do anyway).
Imo this way it is much cleaner. Please double check that `ownProps.scene.route.name` is really what it is used here as.